Code/GraphMol/MolStandardize/Charge.cpp

```diff
@@ -0,0 +1,339 @@
+//
+//  Copyright (C) 2018 Susan H. Leung
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "Charge.h"
+#include <GraphMol/SmilesParse/SmilesParse.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/Substruct/SubstructMatch.h>
+#include <boost/range/adaptor/reversed.hpp>
+
+namespace RDKit {
+namespace MolStandardize {
+
+// The default list of ChargeCorrections.
+std::vector<ChargeCorrection> CHARGE_CORRECTIONS = {
+    ChargeCorrection("[Li,Na,K]", "[Li,Na,K;X0+0]", 1),
+    ChargeCorrection("[Mg,Ca]", "[Mg,Ca;X0+0]", 2),
+    ChargeCorrection("[Cl]", "[Cl;X0+0]", -1)};
+
+// constructor
+Reionizer::Reionizer() {
+	AcidBaseCatalogParams abparams(defaultCleanupParameters.acidbaseFile);
+	this->d_abcat = new AcidBaseCatalog(&abparams);
+	this->d_ccs = CHARGE_CORRECTIONS;
+}
+
+Reionizer::Reionizer(const std::string acidbaseFile) {
+	AcidBaseCatalogParams abparams(acidbaseFile);
+	this->d_abcat = new AcidBaseCatalog(&abparams);
+	this->d_ccs = CHARGE_CORRECTIONS;
+}
+
+Reionizer::Reionizer(const std::string acidbaseFile,	const std::vector<ChargeCorrection> ccs) {
+	AcidBaseCatalogParams abparams(acidbaseFile);
+	this->d_abcat = new AcidBaseCatalog(&abparams);
+	this->d_ccs = ccs;
+}
+
+Reionizer::Reionizer(const Reionizer &other) {
+	d_abcat = other.d_abcat;
```

Comment:
Because you are just assigning the pointer here, and not copying the underlying memory, this is going to cause memory corruption (or errors) if you copy one Reionizer into another and then delete the first.
You need to either copy the underlying object as well, keep track of whether or not you own the pointer, or use smart pointers.